sql/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+        execute("insert into t values (1), (2), (3)");
+    }
+
+    @Test
+    public void testClosePreventsSelect() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow READ operations, " +
+                                        "as it is currently closed.");
+        execute("select * from t");
+    }
+
+    @Test
+    public void testClosePreventsDrop() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow DROP operations, " +
+                                        "as it is currently closed");
+        execute("drop table t");
+    }
+
+    @Test
+    public void testClosePreventsAlter() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow ALTER operations, " +
+                                        "as it is currently closed");
+        execute("alter table t add column x string");
+    }
+
+    @Test
+    public void testClosePreventsRefresh() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow REFRESH " +
+                                        "operations, as it is currently closed.");
+        execute("refresh table t");
+    }
+
+    @Test
+    public void testClosePreventsShowCreate() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow SHOW CREATE " +
+                                        "operations, as it is currently closed.");
+        execute("show create table t");
+    }
+
+    @Test
+    public void testClosePreventsOptimize() throws Exception {
+        prepareClosedTable();
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("The relation \"doc.t\" doesn't support or allow OPTIMIZE " +
+                                        "operations, as it is currently closed.");
+        execute("optimize table t");
+    }
+
+    @Test
+    public void testSelectClosedPartition() throws Exception {
+        execute("create table partitioned_table (i int) partitioned by (i)");
+        ensureYellow();
+        execute("insert into partitioned_table values (1), (2), (3), (4), (5)");
+        refresh();
+        execute("alter table partitioned_table partition (i=1) close");
+        execute("select i from partitioned_table");
+
+        assertEquals(4, response.rowCount());
+    }
+
```

Comment:
I think its worth adding an test for selecting from a closed partition, like: `select * from partitioned_table where i = 1;`